### PR TITLE
note: Rename note driver to noteram driver

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
to reflect this is just a normal note driver with ram buffer

### CODE REVIEW

File: drivers/note/noteram_driver.c [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+ *
+ * Returned Value:
+ *   Zero is returned if the circular buffer is empty.  Otherwise, a negated
+ *   errno value is returned.
+ *
+ ****************************************************************************/
+
+int noteram_register(void)
+{
+  return register_driver("dev/note", &g_noteram_fops, 0666, NULL);
```
Comment (YuuichiNakamura) [inline drivers/note/noteram_driver.c:476]: The first "/" is missing.

File: drivers/note/noteram_driver.c [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+ * Description:
+ *   Register a serial driver at /dev/note that can be used by an
+ *   application to read data from the circular note buffer.
+ *
+ * Input Parameters:
+ *   None.
+ *
+ * Returned Value:
+ *   Zero is returned if the circular buffer is empty.  Otherwise, a negated
+ *   errno value is returned.
```
Comment (YuuichiNakamura) [inline drivers/note/noteram_driver.c:470]: This comment is wrong. Zero is returned if the device is registered successfully.

File: drivers/note/noteram_driver.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/note/noteram_driver.c]: Done.

File: drivers/note/noteram_driver.c (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/note/noteram_driver.c]: Done.

**Comment by liuguo09** [comment]
LGTM. All comments resolved.

**Pull request merged**